frontend/javascripts/dashboard/transfer_task_modal.tsx

```diff
@@ -11,67 +12,52 @@ type Props = {
   onCancel: (...args: Array<any>) => any;
   isOpen: boolean;
 };
-type State = {
-  currentUserIdValue: string;
-};
 
-class TransferTaskModal extends React.PureComponent<Props, State> {
-  state: State = {
-    currentUserIdValue: "",
-  };
+const TransferTaskModal: React.FC<Props> = ({ isOpen, onCancel, annotationId, onChange }) => {
+  const [currentUserIdValue, setCurrentUserIdValue] = useState("");
 
-  async transfer() {
-    const annotationId = this.props.annotationId;
+  const handleSelectChange = useCallback((userId: string) => {
+    setCurrentUserIdValue(userId);
+  }, []);
 
+  const transfer = useCallback(async () => {
     if (!annotationId) {
       throw new Error("No annotation id provided");
     }
 
     try {
-      const updatedAnnotation = await transferTask(annotationId, this.state.currentUserIdValue);
-      this.props.onChange(updatedAnnotation);
+      const updatedAnnotation = await transferTask(annotationId, currentUserIdValue);
+      onChange(updatedAnnotation);
```

Comment:
I think a `setCurrentUserIdValue("")` is needed here as once a task was transferred, and the modal is reopened, the transfer button can directly be used again without anything being selected in the dropdown. Else the dropdown could show that the last user is the one to which the task can be transferred